Consolidate build scripts across all amd64 suffixes

proneetv opened the pull request:
I have made the changes in the directory structure so that I can keep the amd64 `build.sh` script in context to all the amd64 variates while I use docker build command to build containers.
```
amd64/
    amd64/
    amd64-hardened/
    amd64-hardened-nomultilib/
    amd64-nomultilib/
    build.sh
```

Caveat: Now the user cannot run docker build command from inside `amd64/amd64*/` directory since it depends on the `build.sh` script which is outside its context.

Hence, `docker build -f ${arch}${suffix}/Dockerfile .` is the way to keep the `build.sh` in context while building the specific docker container. 

For fixing the `build.sh` script so that is works across all variates, have used the `${arch}` and `${suffix}` variables to generate respective url since they are already being passed in the Dockerfile while executing the build script.

Fixes #24

dastergon commented on amd64/.dockerignore:
```diff
@@ -1,8 +0,0 @@
-*.bz2
```
It seems that you deleted .dockerignore by mistake while moving folders

dastergon commented on amd64/README.md:
````diff
@@ -0,0 +1,8 @@
+to build the docker container:
+
+```shell
+docker build -f amd64/Dockerfile .
````
Since it's a single shell command you can write it with single back ticks in one line, like you did with the `build.sh` name above

dastergon commented:
Hey @proneetv, thanks for the contributions!

I added some comments on your PR.

Can you also write a brief description of your changes in the PR?
You could also enrich the readme doc with more details and examples about build script

proneetv commented:
Hello @dastergon, I have addressed your comments and have force-pushed the changes to avoid unnecessary commits.

Sure, I'll add a description here as well as in the docs in sometime :smiley:

dastergon commented:
@proneetv  Looks good :+1:  

@alunduil any input on this?

ChaosEngine commented:
LGTM too :thumbsup:

proneetv commented:
@ChaosEngine sorry for replying this late!
It works completely fine for me locally, I presume the root folder on [hub.docker](http://hub.docker.com) needs to be modified since you need to get `build.sh` into context of the `docker build` command.

###### The following needs to be ensured otherwise it wont be able to fetch `build.sh` and fail
For building, say amd64-nomultilib docker, 
```shell
cd amd64/
docker build -f amd64-nomultilib/Dockerfile .
```

ChaosEngine commented:
@p8952 sure locally it works nice, but on https://hub.docker.com/ there is no option to change "current working directory" where ```build.sh``` and ```Dockerfile``` is located. Both It need to be located in one directory.
Other than that PR is ok for me.

dastergon commented:
@proneetv any update regarding @ChaosEngine's comments? It would be great to have a unified solution for both local and remote build.

proneetv commented:
@dastergon I can't think of a better approach! :disappointed: 
you have anything in mind? or else we should close this PR!

ChaosEngine commented:
Nope, sorry. Maybe in the future docker will change hub to include this

dastergon commented:
I am closing this PR for now. Issue #24 will remain open though. 
If there's any update that fixes the issue please open a new PR
